src/charm.py

```diff
@@ -513,6 +515,14 @@ def configure_rgw_service(self, event: ops.framework.EventBase) -> None:
                     "rgw_keystone_service_token_accepted_roles": self.id_svc.interface.admin_role,
                     "rgw_s3_auth_use_keystone": str(True).lower(),
                 }
+                namespace_projects = self.leader_get("namespace-projects")
```

Comment:
When does the corresponding `set` operation for `namespace-projects` take place ? As in, who sets this key value.

Reply:
https://github.com/canonical/charm-microceph/blob/main/src/charm.py#L385. The value will be set during deploy time only.

Reply:
perfect. This value won't change throughout the charm's life.